### ISSUE TITLE
Likes and Dislikes

### ISSUE THREAD

**Issue by tmobaird** [cubadomingo/words_and_moves#44]
I saw on the wireframes that each post or event listed had an up/down arrow with a corresponding count. I assumed this would be generated by upvotes.count - downvotes.count. I think the best way to do this would be to create a Like and Dislike or UpVote and DownVote models that map to a user and an event/post. This will easily allow us to enforce that a user can only like/dislike a post once and still allow us to generate this count.

Any ideas?

**Comment by cubadomingo** [comment]
Don't know too much about implementing the voting system, but that sounds good to me.

**Comment by tmobaird** [comment]
it would just be something like this:

```ruby
class User
  has_many :likes
  has_many :dislikes 
end
class Like
  belongs_to :user
  belongs_to :post # belongs to post or event not both
  belongs_to :event
end
class Dislike
  belongs_to :user
  belongs_to :post
  belongs_to :event
end
```

**Comment by cubadomingo** [comment]
What if post and events have a score, and that score contains the likes and dislikes. The score can be adding the likes and subtracting the dislikes. Also the ratio of likes to dislikes could be used to create hot and rising filters.  hot ( +1200, -123) Rising (+450, -23) + age of post or something like that.

**Comment by tmobaird** [comment]
Exactly! That's exactly what I had in mind brotha!

**Comment by tmobaird** [comment]
let me know if you want me to implement the likes/dislikes models

**Comment by cubadomingo** [comment]
Go for it bro! 👍

**Comment by tmobaird** [comment]
Hey brotha, feel free to take a look at the branch I pushed up: adding-likes-and-dislikes. It's got the absolute basic functionality for liking and disliking. It's working pretty nicely so far, I'll just need to make sure that a user can't like/dislike something more than once. I also made the association polymorphic for the likes event/post that it belongs to. So basically a like/dislike belongs to an item which is either and event or post
